core/src/main/java/de/bwaldvogel/mongo/backend/FieldUpdates.java

```diff
@@ -343,8 +343,15 @@ private void handleRename(String key, Object toField) {
 
     private void applyRenames() {
         for (Entry<String, String> entry : renames.entrySet()) {
+            if (!Utils.canFullyTraverseSubkeyForRename(document, entry.getKey())) {
+                throw new MongoServerException("cannot traverse element");
```

Comment:
I think to be closer to real MongoDB we should rather throw an `PathNotViableException` here
```suggestion
                throw new PathNotViableException("cannot traverse element");
```